NString.Tests/StringTemplateConverterTests.cs

```diff
@@ -0,0 +1,335 @@
+﻿using System.Globalization;
+using System.Linq;
+using Xunit;
+// ReSharper disable InconsistentNaming
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace NString.Tests
+{
+    public class StringTemplateConverterTests
+    {
+        private static readonly string[] StringCollectionValues = { "1", "2", "3" };
+        private const string ExpectedStringCollectionConcatenation = "1|2|3";
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Property()
+        {
+            var values = new PropertyOrFieldValues { ConcatenatedStringCollectionProperty = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{ConcatenatedStringCollectionProperty}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Field()
+        {
+            var values = new PropertyOrFieldValues { ConcatenatedStringCollectionField = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{ConcatenatedStringCollectionField}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Reference_Type()
```

Comment:
Unless I'm missing something, this test doesn't cover anything that's not already covered by the first test?

Reply:
You are right. I wanted to have explicit tests for property/field accesses, and reference/value types, but they happen to be duplicate.

The test is pretty basic and is probably covered indirectly by other tests as well.  Should I remove it?

Reply:
Yes please!